core/trino-main/src/main/java/io/trino/dispatcher/DispatchManager.java

```diff
@@ -236,6 +241,7 @@ private <C> void createQueryInternal(QueryId queryId, Span querySpan, Slug slug,
             Optional<String> preparedSql = Optional.ofNullable(preparedQuery).flatMap(PreparedQuery::getPrepareSql);
             DispatchQuery failedDispatchQuery = failedDispatchQueryFactory.createFailedDispatchQuery(session, query, preparedSql, Optional.empty(), throwable);
             queryCreated(failedDispatchQuery);
+            queryMonitor.queryImmediateFailureEvent(failedDispatchQuery.getBasicQueryInfo(), toFailure(throwable));
```

Comment:
LGTM but add a comment whgy this is not happening within `createFailedDispatchQuery` for future readers.

Reply:
Comment added. Let me know if this is what you meant and whether I can do anything else to get an approval?